integration_tests/testsuite/driver.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python
+
+# Copyright 2016 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import sys
+
+import deploy_app
+import test_exception
+import test_logging
+import test_monitoring
+import test_root
+
+# TODO (nkubala): make this a configurable parameter from cloudbuild
+# required to be paired with '--no-deploy'
+DEFAULT_URL = 'https://nick-cloudbuild.appspot.com'
+
+
+def _main():
+    logging.getLogger().setLevel(logging.DEBUG)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--image', '-i', help='Newly-constructed base \
```

Comment:
`image` and `directory` aren't always required (if you're not deploying the app then you don't need them). I'm checking that these arguments exist when they should in the deploy logic below the argparse; is this ok in place of using the required flag? This way I can give a more descriptive error message to the user (instead of letting argparse just throw an error).